### ISSUE TITLE
sample_adj function

### ISSUE THREAD

**Issue by k-s-b** [rusty1s/pytorch_sparse#148]
Where can I find the implementation for `torch.ops.torch_sparse.sample_adj` function, which is called in `sample_adj` function defined [here](https://github.com/rusty1s/pytorch_sparse/blob/8d03163af730bdb7d7a115ce1f2e3627f036cb1a/torch_sparse/sample.py)? 
I can not find the module `torch.ops.torch_sparse` . I am able to run the models that use the `sample_adj` function, so I think I have installed everything correctly.

**Comment by rusty1s** [comment]
You can find it [here](https://github.com/rusty1s/pytorch_sparse/blob/master/csrc/cpu/sample_cpu.cpp), which gets loaded into PyTorch [here](https://github.com/rusty1s/pytorch_sparse/blob/master/torch_sparse/__init__.py#L10-L15).

**Comment by k-s-b** [comment]
Got it. Thanks for the prompt reply!

**Comment by Antoine-ls** [comment]
@rusty1s I want to control max multi-hop neighbors of each node, I am now using sample_adj to sample 10 edges for each node, but I find that the sampled adj is not symmetric,  because of this asymmetry, I find that when the sampling is done, I use k_hop_subgraph to calculate 3 hop neighbors for certain nodes, involved nodes are more than 1000, while I think the maximal 3 hop neighbors are 1000. 
Is there a way to sample while ensuring symmetry or do you have any suggestions for me do do the max multi-hop neighbors control?

Here is the example code
adj,_ = adj.sample_adj(torch.arange(adj.sizes()[0]),sampled_edges)
row, col, edge_idx = adj.coo()
data.edge_index = torch.stack([row, col], dim=0)
...
two_hop_nodes,_,_,_ = k_hop_subgraph(i, 3, data.edge_index)

**Comment by rusty1s** [comment]
You can probably use the `n_id` output of `sample_adj` to get the full sub-graph of sampled nodes:
```python
_, n_id = adj.sample_adj(torch.arange(adj.sizes()[0]),sampled_edges)

edge_index, _ = subgraph(n_id, edge_index, relabel_nodes=True)
```
